File: 0204 count-primes/solution.java
class Solution {
    public int countPrimes(int n) {
        if(n <= 2) return 0;
        boolean[] isPrime = new boolean[n];
        // Arrays.fill(isPrime, true);
        isPrime[0] = true;
        isPrime[1] = true;
        for(int i = 2; i*i <= n; i++){
            if(!isPrime[i]){
                for(int mul = i*i; mul<n; mul+=i){
                    isPrime[mul] = true;
                }
            }
        }
        int res = 0;
        for(boolean b: isPrime){
            if(!b) res++;
        }
        return res;
    }
}

File: 0204 count-primes/solution.cs
public class Solution {
    public int CountPrimes(int n) {
        if(n<=2) return 0;
        bool[] pri=new bool[n];
        pri[0]=pri[1]=true;
        int c=0;
        for(int i=2;i*i<n;i++) {
            if(!pri[i]) {
                 for(int j=i*i;j<n;j+=i) {
                 pri[j]=true;
            }
            }
        }
        foreach(bool b in pri) if(!b) c++;
        return c;
    }
}
